### PR TITLE
Move new runs page into shared lib

### PR DESCRIPTION
**Pull request by goodoldneon**
## Description
Move new runs page into shared lib. There should be 0 functional changes: the page should behave exactly the same

## Type of change (choose one)
- [x] Chore (refactors, upgrades, etc.)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [ ] I've linked any associated issues to this PR.
- [ ] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

File: ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/runs/page.tsx [hunk 1]
```diff
@@ -225,49 +195,23 @@ export default function RunsPage({
...
+      features={{
+        history: features.data?.history ?? 7,
+      }}
+      functionSlug={functionSlug}
+      hasMore={hasNextPage ?? false}
+      isLoadingInitial={firstPageRes.isLoading}
+      isLoadingMore={nextPageRes.isLoading}
+      getRun={getRun}
+      onScroll={fetchMoreOnScroll}
+      onScrollToTop={console.log}
```
Comment (anafilipadealmeida) [inline ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/runs/page.tsx:210]: Will we need this one?

File: ui/packages/components/src/types/features.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export type Features = {
+  history: number;
```
Comment (anafilipadealmeida) [inline ui/packages/components/src/types/features.ts:2]: Is this a number?

**Review by anafilipadealmeida — APPROVED** [review]

File: ui/packages/components/src/types/features.ts (hunk 2 above)
Comment (goodoldneon) [reply ui/packages/components/src/types/features.ts:2]: Yea:
https://github.com/inngest/inngest/blob/main/ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/%5BenvironmentSlug%5D/functions/%5Bslug%5D/runs/TimeFilter.tsx#L39-L41

File: ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/runs/page.tsx (hunk 1 above)
Comment (goodoldneon) [reply ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/runs/page.tsx:210]: Oops!

**Pull request merged**